node/src/command.rs

```diff
@@ -31,21 +32,68 @@ enum ChainIdentity {
 	Development,
 }
 
+impl ChainIdentity {
```

Comment:
I think that the code is cleaner by adding some specific `ChainIdentity` based logic and `match` expressions to the `ChainIdentity` enum itself. This way some of the `match` expressions in the rest of the code can be removed.

Furthermore I unified the code in the previous `identify` and `load_spec` function which was strangely different: whereas one tested for string equality, the other one tested only whether initial fragments of string are equal. Testing for equality is enough as the only `id`s we deal with are the ones created in the file `chain_spec` or that already exist in any of our precomputed chain spec files in the folder `res`.